packages/devtools_app/lib/src/screens/memory/panes/diff/controller/diff_pane_controller.dart

```diff
@@ -36,28 +35,27 @@ class DiffPaneController {
   DiffListItem get selectedItem => snapshots.value[selectedIndex.value];
 
   /// Full name for the selected class.
-  ValueListenable<String?> get selectedClass => _selectedClass;
-  final _selectedClass = ValueNotifier<String?>(null);
-  void setSelectedClass(String? value) => _selectedClass.value = value;
+  final selectedClassName = ValueNotifier<String?>(null);
```

Comment:
Whats the point to expose notifier instead of setter here?

Reply:
This allows the Table code to set the value of the notifier when a row is selected. Otherwise we have to pass a bunch of callbacks around that set the value of the notifiers, and in order to have selection for a table wired up properly, we'd have to pass two parameters: selectionNotifier and onSelected (where on selected just calls selectionNotifier.value = value)